### DISCUSSION TITLE
디버깅 원칙 | kciter.so

### DISCUSSION THREAD

**Discussion by giscus** [kciter/kciter.github.io#34] [Announcements]
# 디버깅 원칙 | kciter.so

필자를 포함하여 대부분의 개발자가 만드는 것은 생명과 직결된 항공이나 핵 발사 프로그램 같은 것은 아닐 것이다. 따라서 우리가 만드는 제품은 생산성과 안정성 사이에서 생산성을 택하는 경우가 많으며 이로 인해 발생하는 품질 확인 공백은 결함으로…

https://kciter.so/posts/principles-of-debugging/

**Comment by sappho192** [comment]
좋은 글 감사합니다.
평소에 단편적으로, 막연하게만 생각하던 것들이 이렇게 체계적으로 정리되어 있어서 저도 생각을 정리하는데에 많은 도움이 되었습니다.

개인적인 노하우로는 실험 단계에서 코드를 작성할 때, 기존의 솔루션과는 별개로 그 문제와 연관된 부분들만 구현되어있는 최소한의 예시 코드를 만들어 실험을 진행하는 것이 있습니다.
규모가 어느정도 커진 솔루션에서는 다른 부분과 서로 얽혀있는게 많아서, 의심가는 특정 부분만 떼어서 이리저리 바꿔보기 힘든 경우가 많더라구요.
그래서 minimal-example에서 원인을 파악하고, 해결책을 만들어낸 다음, 기존 솔루션에서 문제가 되는 부분에 반영하여 해결하곤 했습니다.

**Reply by kciter** [reply]
좋게 봐주셔서 감사합니다. 🙂

저도 인프라 관련 문제를 해결할 때 최대한 비슷한 환경을 구축해서 실험했던 기억이 떠오르네요. 좋은 노하우를 공유해주셔서 감사합니다.

**Comment by shimdokite** [comment]
안녕하세요~! 6개월차 개발자입니다.
실무를 하면서 느낀 점은 기능 구현만큼 회사의 레거시 코드들이나 내가 작성한 코드에 나타나는 버그를 수정할 때가 더 많다는 점이었습니다.
확실히 에러를 접하는 빈도가 많아지면서 점차 저만의 디버깅 방법이 구축되고, 동료분들 덕분에 많이 배우고 있습니다!
그렇지만 아직은 경험이 부족해 작은 직관이 부족하고, 원인이 있는 지점을 빠르게 찾거나, 다른 가능성을 생각하는 게 어려운 점 등 부족함도 느껴졌습니다.
정성스럽게 작성해 주신 글을 읽고 저와 대조해 보면서 잘하고 있는 부분은 계속 가져가고, 적용해 볼 수 있는 부분은 꼭 적용해 봐야겠다는 생각이 들었습니다.
좋은 글 작성해 주셔서 감사합니다!☺️

**Reply by kciter** [reply]
좋게 봐주셔서 감사합니다. 😄 앞으로도 이 글이 많은 도움이 됐으면 좋겠네요!

**Comment by euijinkk** [comment]
암묵지에 있는 과정을 체계적으로 분류해주셔서 도움이 많이 되었습니다! 감사합니다.
